internal/resources/openshift.go

```diff
@@ -0,0 +1,675 @@
+			m.logger.Info("Metrics Role already exists", "key", roleKey)
+		} else {
+			m.logger.Error(err, "Failed to create Metrics Role",
+				"key", roleKey)
+		}
+		return nil, false, err
+	}
+	return roleWant, true, nil
+}
+
+func (m *OpenShiftManager) getMetricsRole(
+	ctx context.Context) (*rbacv1.Role, types.NamespacedName, error) {
+	// ---
+	roleKey := m.metricsRoleNsname()
+	role := &rbacv1.Role{}
+	err := m.client.Get(ctx, roleKey, role)
+	if err != nil {
+		return nil, roleKey, err
+	}
+	return role, roleKey, nil
+}
+
+func (m *OpenShiftManager) getOrCreateMetricsRoleBinding(
+	ctx context.Context,
+	role *rbacv1.Role) (*rbacv1.RoleBinding, bool, error) {
+	// ---
+	roleBindCurr, roleBindKey, err := m.getMetricsRoleBinding(ctx)
+	if err == nil {
+		return roleBindCurr, false, err // OK
+	}
+	if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get Metrics RoleBinding",
+			"key", roleBindKey)
+		return nil, false, err
+	}
+	roleBindWant := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: roleBindKey.Namespace,
+			Name:      roleBindKey.Name,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "prometheus-k8s",
+				Namespace: "openshift-monitoring",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "Role",
+			Name: role.GetName(),
+		},
+	}
+	err = m.client.Create(ctx, roleBindWant, &rtclient.CreateOptions{})
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			m.logger.Info("Metrics RoleBinding already exists",
+				"key", roleBindKey)
+		} else {
+			m.logger.Error(err, "Failed to create Metrics RoleBinding",
+				"key", roleBindKey)
+		}
+		return nil, false, err
+	}
+	return roleBindWant, true, nil
+}
+
+func (m *OpenShiftManager) getMetricsRoleBinding(
+	ctx context.Context) (
+	*rbacv1.RoleBinding, types.NamespacedName, error) {
+	// ---
+	roleBindKey := m.metricsRoleBindingNsname()
+	roleBind := &rbacv1.RoleBinding{}
+	err := m.client.Get(ctx, roleBindKey, roleBind)
+	if err != nil {
+		return nil, roleBindKey, err
+	}
+	return roleBind, roleBindKey, nil
+}
+
+func (m *OpenShiftManager) metricsRoleNsname() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: m.cfg.PodNamespace,
+		Name:      metricsRoleName,
+	}
+}
+
+func (m *OpenShiftManager) metricsRoleBindingNsname() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: m.cfg.PodNamespace,
+		Name:      metricsRoleBindingName,
+	}
+}
+
+// IsOpenShiftCluster checks if operator is running on OpenShift cluster based
+// on its self-pod annotations.
+func IsOpenShiftCluster(ctx context.Context,
+	reader rtclient.Reader,
+	cfg *conf.OperatorConfig) bool {
+	key := rtclient.ObjectKey{
+		Namespace: cfg.PodNamespace,
+		Name:      cfg.PodName,
+	}
+	clusterType, err := resolveClusterTypeByPod(ctx, reader, key)
+	return (err == nil) && (clusterType == ClusterTypeOpenshift)
+}
+
+// resolveClusterTypeByPod finds the kind of K8s cluster via anbnotation of one
```

Comment:
typo: "anbnotation" -> "annotation"

Reply:
Thanks -- for some reason vscode spell-checker did not catch this one.